src/Xamarin.Android.Build.Utilities/Sdks/AndroidSdkBase.cs

```diff
@@ -83,11 +83,15 @@ public virtual void Initialize (string androidSdkPath = null, string androidNdkP
 			}
 			// we need to look for extensions other than the default .exe|.bat
 			// google have a habbit of changing them.
-			Adb = GetExecutablePath (AndroidPlatformToolsPath, Adb);
-			Android = GetExecutablePath (AndroidToolsPath, Android);
-			Emulator = GetExecutablePath (AndroidToolsPath, Emulator);
-			Monitor = GetExecutablePath (AndroidToolsPath, Monitor);
-			NdkStack = GetExecutablePath (AndroidNdkPath, NdkStack);
+			if (!string.IsNullOrEmpty (AndroidPlatformToolsPath))
+				Adb = GetExecutablePath (AndroidPlatformToolsPath, Adb);
```

Comment:
These changes don't make sense to me.

`GetExecutablePath()` will return `exe` if `Path.Combine(dir, exeWithChangedExtension)` doesn't exist. If e.g. `AndroidPlatformToolsPath` is empty, then e.g. `Path.Combine("", "adb")` will return `"adb"`, which won't exist, so `GetExecutablePath("", Adb)` will return Adb, i.e. the value of `Adb` will be *unchanged*.

(There's one problem here in that `Path.Combine(null, ...)` will throw `ArgumentNullException`, but this can be addressed within `GetExecutablePath()`.)